Tests/GRPCTests/AsyncAwaitSupport/AsyncClientTests.swift

```diff
@@ -394,6 +398,29 @@ final class AsyncClientCancellationTests: GRPCTestCase {
       return .bidirectionalStreaming(echo.makeUpdateCall())
     }
   }
+
+  func testConnectionFailureCancelsRequestStreamWithError() async throws {
+    let echo = try self.makeClient(port: 0) {
+      // Configure a short wait time; we will not start a server so fail quickly.
+      $0.connectionPool.maxWaitTime = .milliseconds(10)
+    }
+
+    do {
+      let update = echo.makeUpdateCall()
+      try await update.requestStream.send(.init())
+      XCTFail("Expected an error to be thrown")
+    } catch {
+      XCTAssertFalse(error is CancellationError)
+    }
+
+    do {
+      let collect = echo.makeCollectCall()
+      try await collect.requestStream.send(.init())
+      XCTFail("Expected an error to be thrown")
+    } catch {
+      XCTAssertFalse(error is CancellationError)
+    }
```

Comment:
Nit: Should we add the [overload](https://github.com/swift-server/async-http-client/blob/1af18d2f95a24358b9e0001d9597a4646d2555ca/Tests/AsyncHTTPClientTests/XCTest%2BAsyncAwait.swift#L64-L77) from AHC to gRPC as well?

Reply:
No, we should just use the one that AHC copied from gRPC 🙈